tests/IdentifyServiceProviderTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Unicodeveloper\Identify\Test;
+
+use Mockery\MockInterface;
+use PHPUnit_Framework_TestCase;
+use Unicodeveloper\Identify\IdentifyServiceProvider;
+
+class IdentifyServiceProviderTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var MockInterface
+     */
+    protected $mockApp;
+
+    /**
+     * @var IdentifyServiceProvider
+     */
+    protected $provider;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->mockApp = \Mockery::mock();
+        $this->provider = new IdentifyServiceProvider($this->mockApp);
+    }
+
+    /**
+     * Test that it provides the correct service name
+     */
+    public function testRegister()
+    {
+        $this->mockApp->shouldReceive('bind')->once()->withAnyArgs()->once();
+
+        $this->provider->register();
+    }
```

Comment:
Okay,
Thanks for the pointers, I'll update and push.

With respect to Mocks, I'm used to working with Anonymous Mocks and it
works, so I've not really seen the purpose of binding to a Class.

But I see your point now, and I'll dive deeper into Mockery to actually
understand it.
On Apr 21, 2016 6:56 AM, "Prosper Otemuyiwa" <notifications@github.com>
wrote:

> In tests/IdentifyServiceProviderTest.php
> <https://github.com/unicodeveloper/laravel-identify/pull/4#discussion_r60529437>
> :
>
> > +    protected function setUp()
> > +    {
> > +        parent::setUp();
> > +        $this->mockApp = \Mockery::mock();
> > +        $this->provider = new IdentifyServiceProvider($this->mockApp);
> > +    }
> > +
> > +    /**
> > +     * Test that it provides the correct service name
> > +     */
> > +    public function testRegister()
> > +    {
> > +        $this->mockApp->shouldReceive('bind')->once()->withAnyArgs()->once();
> > +
> > +        $this->provider->register();
> > +    }
>
> Check this out for testing the register method of the Service provider.
>
> Mind you our $this->mockApp would no longer be an anonymous mock. It will
> mock the ServiceProvider in the setUp method like so:
>
> $this->mockApp =
> m::mock('Unicodeveloper\Identify\IdentifyServiceProvider', [$this->app]);
>
> We are also mocking the Laravel Ioc Container itself with $this->app =
> m::mock('App');
>
> $this->app->shouldReceive('bind')          ->once()          ->andReturnUsing(                function ($name, $closure) use ($test) {                    $test->assertEquals('laravel-identify', $name);                    $test->assertInstanceOf(                        'Unicodeveloper\Identify\Identify',                        $closure()                    );                }            );$this->mockApp->register();
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/unicodeveloper/laravel-identify/pull/4/files/742ff9c22c2386b6cf86e08753a26bea523fd54a#r60529437>
>